components/src/components/shared/checkout-dialog/CheckoutDialog.tsx

```diff
@@ -175,19 +166,28 @@ export const CheckoutDialog = ({ top = 0 }: CheckoutDialogProps) => {
 
   useEffect(() => {
     setAddOns((prevAddOns) => {
```

Comment:
I suspect that this `prevAddOns` will never get reset to the original list. This means we will keep removing add ons, but never add any back if the user returns to the first step in the flow, and selects a different plan.